plugins/module_utils/network/sonic/facts/ospfv2/ospfv2.py

```diff
@@ -0,0 +1,321 @@
+    remove_empties_from_list
+)
+from ansible.module_utils.connection import ConnectionError
+
+
+network_instance_path = '/data/openconfig-network-instance:network-instances/network-instance'
+protocol_ospf_path = 'protocols/protocol=OSPF,ospfv2/ospfv2'
+
+
+class Ospfv2Facts(object):
+    """ The sonic ospfv2 fact class
+    """
+
+    def __init__(self, module, subspec='config', options='options'):
+        self._module = module
+        self.argument_spec = Ospfv2Args.argument_spec
+        spec = deepcopy(self.argument_spec)
+        if subspec:
+            if options:
+                facts_argument_spec = spec[subspec][options]
+            else:
+                facts_argument_spec = spec[subspec]
+        else:
+            facts_argument_spec = spec
+
+        self.generated_spec = utils.generate_dict(facts_argument_spec)
+
+    def populate_facts(self, connection, ansible_facts, data=None):
+        """ Populate the facts for ospfv2
+        :param connection: the device connection
+        :param ansible_facts: Facts dictionary
+        :param data: previously collected conf
+        :rtype: dictionary
+        :returns: facts
+        """
+        objs = []
+
+        all_ospfv2_configs = {}
+
+        if not data:
+            all_ospfv2_configs = self.get_ospfv2(self._module)
+
+        for ospf_config in all_ospfv2_configs:
+            if ospf_config:
+                obj = self.render_config(self.generated_spec, ospf_config)
+                if obj:
+                    objs.append(obj)
+
+        ansible_facts['ansible_network_resources'].pop('ospfv2', None)
+        facts = {}
+        if objs:
+            params = utils.validate_config(self.argument_spec, {'config': objs})
+            facts['ospfv2'] = remove_empties_from_list(params['config'])
+        ansible_facts['ansible_network_resources'].update(facts)
+        return ansible_facts
+
+    def render_config(self, spec, conf):
+        """
+        Render config as dictionary structure and delete keys
+          from spec for null values
+
+        :param spec: The facts tree, generated from the argspec
+        :param conf: The configuration
+        :rtype: dictionary
+        :returns: The generated config
+        """
+        return conf
+
+    def get_ospfv2(self, module):
+        """Get all OSPFv2 configurations available in chassis"""
+        ospf_configs = []
+        vrfs = get_all_vrfs(module)
+        for vrf_name in vrfs:
+            all_ospf_path = '%s=%s/%s' % (network_instance_path, vrf_name, protocol_ospf_path)
+            request = [{"path": all_ospf_path, "method": "GET"}]
+
+            try:
+                response = edit_config(module, to_request(module, request))
+            except ConnectionError as exc:
+                module.fail_json(msg=str(exc), code=exc.code)
+
+            for resp in response:
+                if 'openconfig-network-instance:ospfv2' in resp[1]:
```

Comment:
Since only one response is received, the data can be directly accessed by:
```suggestion
            if 'openconfig-network-instance:ospfv2' in resp[0][1]:
```